src/pages/sicp/subcomponents/chatbot/ChatBox.tsx

```diff
@@ -22,6 +22,11 @@ const BOT_ERROR_MESSAGE: Readonly<ChatMessage> = {
   role: 'assistant'
 };
 
+const MESSAGE_TOO_LONG_MESSAGE: Readonly<ChatMessage> = {
+  content: 'Your message is too long. Please try again with a shorter message.',
+  role: 'assistant'
+};
```

Comment:
That way, this will never be necessary. From UX perspective, it's easier for the user to edit their message before sending in order to fit the character limit, as opposed to having to copy and paste their previous message and send it again. Not to mention, if you do it like this, there will be a mismatch between the conversation history stored in the FE state with the one stored in the DB

Reply:
Hi, I think it's better to stay with the current design. By not reseting "user input" to empty string, use can still edit their message as you suggested;

This would be similar to the behavior of chatGPT bot. If the message is too long, after sending the message to chatGPT, user receives an error message and the input field remains the same. So I find this a more intuitive behavior.

I feel it is OK for the mismatch between the conversation history as blocking messages that are too long before it reaching backend is the purpose of this feature.

One customization I made is to not show the "overly long" message in the chat field so it is less painful to scroll up and refer to earlier conversation.

Reply:
You are right; 
it is more intuitive to do let the user trigger this in the first place;
this also pollute the chat history, making it less readable.